spec/webvalve/service_url_converter_spec.rb

```diff
@@ -0,0 +1,141 @@
+require 'spec_helper'
+
+RSpec.describe WebValve::ServiceUrlConverter do
+  let(:url) { "http://bar.com" }
+
+  subject { described_class.new(url: url) }
+
+  describe '#regexp' do
+    it "returns a regexp" do
+      expect(subject.regexp).to be_a(Regexp)
+    end
+
+    context "with a trailing *" do
+      let(:url) { "http://bar.com/*" }
+
+      it "matches when empty" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches when existing" do
+        expect("http://bar.com/foobaloo").to match(subject.regexp)
+      end
+
+      it "matches with additional tokens" do
+        expect("http://bar.com/foobaloo/wink").to match(subject.regexp)
+      end
+
+      it "doesn't match when missing the trailing slash tho" do
+        expect("http://bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a totally wildcarded protocol" do
+      let(:url) { "*://bar.com" }
+
+      it "matches http" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches anything else" do
+        expect("gopher://bar.com/").to match(subject.regexp)
+      end
+    end
+
+    context "with a wildcarded partial protocol" do
+      let(:url) { "http*://bar.com" }
+
+      it "matches empty" do
+        expect("http://bar.com/").to match(subject.regexp)
+      end
+
+      it "matches full" do
+        expect("https://bar.com/").to match(subject.regexp)
+      end
+    end
+
+    context "with a TLD that is a substring of another TLD" do
+      let(:url) { "http://bar.co" }
+
+      it "doesn't match a different TLD when extending" do
+        expect("http://bar.com").not_to match(subject.regexp)
+      end
+    end
+
+    context "with a wildcard subdomain" do
+      let(:url) { "http://*.bar.com" }
+
+      it "matches" do
+        expect("http://foo.bar.com").to match(subject.regexp)
+      end
+
+      it "doesn't match when too many subdomains" do
+        expect("http://beep.foo.bar.com").not_to match(subject.regexp)
```

Comment:
I think this is an acceptable level of flex. You can't do arbitrary subdomain depth wildcarding, `**`-style. I could actually implement `**` if we wanted to, but seems unneccessary.

Reply:
it looks like we don't have a test case for when the url includes multiple `*`s, like `**`. what does it do when we do that? is that behavior correct? safe?

Reply:
I think it'd just match the same as a single * but I'll throw some tests at it

Reply:
Yeah, `**` acts the same as `*`, threw two tests demonstrating this in.